pkg/ddl/reorg.go

```diff
@@ -706,7 +706,10 @@ func getReorgInfo(ctx *JobContext, d *ddlCtx, rh *reorgHandler, job *model.Job,
 			tb = tbl.(table.PhysicalTable)
 		}
 		if mergingTmpIdx {
-			start, end = tablecodec.GetTableIndexKeyRange(pid, tablecodec.TempIndexPrefix|elements[0].ID)
+			firstElemTempID := tablecodec.TempIndexPrefix | elements[0].ID
```

Comment:
Does all multi-schema change index with continuous id？

Reply:
They are always increasing, but not always continuous: if DDL adds another index between these indexes, it is possible that:

```
multi-schema add index IDs = [1, 2, 4, 5]
another index ID = [3]
```

If the new index rollbacked, temp index 0xFFFFxxx03 may have dirty records. I think we need to skip these dirty records.

Reply:
[01be552](https://github.com/pingcap/tidb/pull/51747/commits/01be5523d88c59cb0f2e73278fb24dc916c790ac)

Reply:
Agree